src/compas_robots/model/robot.py

```diff
@@ -990,6 +1004,181 @@ def _check_link_name(self, name):
         if name in all_link_names:
             raise ValueError("Link name '%s' already used in chain." % name)
 
+    # --------------------------------------------------------------------------
+    # Methods for accessing the visual and collision geometry
+    # --------------------------------------------------------------------------
+
+    def _extract_link_meshes(self, link_elements, meshes_at_link_origin=True):
+        # type: (List[Visual] | List[Collision], Optional[bool]) -> List[Mesh]
+        """Extracts the list of compas meshes from a link's Visual or Collision elements.
+
+        Parameters
+        ----------
+        link_elements : list of :class:`~compas_robots.model.Visual` or :class:`~compas_robots.model.Collision`
+            The list of Visual or Collision elements of a link.
+        meshes_at_link_origin : bool, optional
+            Defaults to True, which means that the meshes will be transformed according to
+            `element.origin`, such that the mesh origin matches with the link's origin frame.
+            If False, the meshes will be extracted as it is loaded from the
+            robot model package. Note that the `.origin` for each element is not necessarily
+            the same.
+
+        Returns
+        -------
+        list of :class:`~compas.datastructures.Mesh`
+            A list of meshes belonging to the link elements.
+
+        Notes
+        -----
+        Only MeshDescriptor in `element.geometry.shape` is supported. Other shapes are ignored.
+
+        """
+        if not link_elements:
+            return None
+
+        meshes = []
+        # Note: Each Link can have multiple visual nodes
+        for element in link_elements:
+            # Some elements may have a non-identity origin frame
+            origin = element.origin.to_compas_frame() if element.origin else Frame.worldXY()
+            t_origin = Transformation.from_frame(origin)
```

Comment:
here you could also just give `element.origin` directly to `Transformation.from_frame()` making the `to_compas_frame` unnecessary. `FrameProxy` should behave at-least like a `Frame`.

Reply:
Yes, agree with you,   replaced with 
```t_origin = Transformation.from_frame(element.origin or Frame.worldXY())```